timefold-solver-python-core/src/main/java/ai/timefold/solver/python/DaemonThreadFactory.java

```diff
@@ -0,0 +1,15 @@
+package ai.timefold.solver.python;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+
+public class DaemonThreadFactory implements ThreadFactory {
```

Comment:
@Christopher-Chianelli We've just had a very long conversation about why this is necessary. This conversation should have been a comment in this class. ;-) Please add it.

Explain the catch 22, explain why we can't free the resources on our own, explain the consequences. 

Ideally, the Python solver manager config API in Python should warn users that, if they want to override this, they have to do their own daemon threads otherwise their processes are going to hang.

Reply:
`SolverManagerConfig` in Python does not have `threadFactoryClass` since Python does not have a counterpart.